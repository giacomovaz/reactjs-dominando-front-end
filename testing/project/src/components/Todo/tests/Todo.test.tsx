import { render, screen } from "@testing-library/react";
import { Todo } from "..";
import userEvent from "@testing-library/user-event";

describe("<Todo />", () => {
  it("matches the style", () => {
    const { container } = render(<Todo />);

    expect(container).toMatchSnapshot();
  });

  it("renders the input", () => {
    render(<Todo />);

    expect(screen.getByPlaceholderText("Type your todo")).toBeInTheDocument();
  });

  it("renders the submit button", () => {
    render(<Todo />);

    expect(screen.getByRole("button", { name: "Add" })).toBeInTheDocument();
  });

  describe("when the input is filled", () => {
    describe("and the submit button is clicked", () => {
      it("adds a new item", () => {
        render(<Todo />);

        const input = screen.getByPlaceholderText("Type your todo");
        userEvent.type(input, "Buy milk");

        const submitButton = screen.getByRole("button", { name: "Add" });
        userEvent.click(submitButton);

        expect(screen.getByRole("list")).toHaveTextContent("Buy milk");
      });

      it("clear the input", () => {
        render(<Todo />);

        const input = screen.getByPlaceholderText("Type your todo");
        userEvent.type(input, "Buy milk");

        const submitButton = screen.getByRole("button", { name: "Add" });
        userEvent.click(submitButton);

        expect(screen.getByPlaceholderText("Type your todo")).toHaveTextContent(
          ""
        );
      });
    });
  });
});
