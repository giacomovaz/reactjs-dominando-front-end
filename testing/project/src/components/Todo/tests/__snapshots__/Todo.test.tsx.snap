// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> matches the style 1`] = `
<div>
  <div>
    <form>
      <input
        placeholder="Type your todo"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
    <ul />
  </div>
</div>
`;
